scopes/component/component/ui/menu/menu.tsx

```diff
@@ -66,29 +72,28 @@ function VersionRelatedDropdowns({ component }: { component: ComponentModel }) {
       [component.tags]
     ) || [];
 
-  const isLatestVersion = useMemo(() => component.version === versionList[0], [component.version]);
-  const packageVersion = useMemo(() => (isLatestVersion ? '' : `@${component.version}`), [component.version]);
-  const origin = typeof window !== undefined ? window.location.origin : undefined;
-  const finalElementsUrl = origin && component.elementsUrl ? `${origin}${component.elementsUrl}` : undefined;
-
+  const methods = getConsumeMethodsComponents(consumeMethods, component);
   return (
     <>
       {versionList.length > 0 && (
         <UseBoxDropdown
           position="bottom-end"
           className={styles.useBox}
-          Menu={() => (
-            <UseBoxMenu
-              componentName={component.id.name}
-              componentId={component.id.toString({ ignoreVersion: isLatestVersion })}
-              packageName={`${component.packageName}${packageVersion}`}
-              elementsUrl={finalElementsUrl}
-              registryName={component.packageName.split('/')[0]}
-            />
-          )}
+          Menu={<ConsumeMethodsMenu methods={methods} componentName={component.id.name} />}
         />
       )}
       <VersionDropdown versions={versionList} currentVersion={component.version} />
     </>
   );
 }
+
+function getConsumeMethodsComponents(consumeMethods: OrderedConsumeSlot, componentModel: ComponentModel) {
```

Comment:
I'd rename it - `useConsumeMethodSlot(consumeMethodsSlot, component)`
it's a hook, people might break the Rules Of Hooks, so it's important to call it `useXXX`